src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -296,6 +296,18 @@ await _transport.SendMessageAsync(new JsonRpcResponse
         }, cancellationToken).ConfigureAwait(false);
     }
 
+    private void RegisterCancellation(CancellationToken cancellationToken, RequestId requestId)
+    {
+        cancellationToken.Register(async () => await SendMessageAsync(new JsonRpcNotification
```

Comment:
Register takes an Action. The way this is written, it's creating an async void method. It should instead do something like `() => _ = SendMessageAsync(...)`.